src/rise-lisk/commands/txs/registerDelegateTx.c

```diff
@@ -8,10 +8,11 @@
 #include "../../../io.h"
 #include "../../../ui_utils.h"
 
-static char username[21];
-static uint8_t read;
-static uint16_t totalLengthAfterAsset;
+#define USERNAME_MAX_LEN 20
 
+static char username[USERNAME_MAX_LEN];
```

Comment:
its 20+\n ?

Reply:
No, it's pure data. There's no point in pretending that it's a C-string (`\n` terminator), as the logic below just fills it up with the username and the keys data, and later determines where the username ends. The `\n` terminator is added later when it's copied over to the `lineBuffer`.